src/PeriodEnd/Refunds/SFA.DAS.ProviderPayments.Calc.Refunds.IntegrationTests/Infrastructure/GivenAPaymentsHistoryRepository.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using NUnit.Framework;
+using SFA.DAS.ProviderPayments.Calc.Refunds.Infrastructure.Entities;
+using SFA.DAS.ProviderPayments.Calc.Refunds.Infrastructure.Repositories;
+using SFA.DAS.ProviderPayments.Calc.Refunds.IntegrationTests.Attributes;
+
+namespace SFA.DAS.ProviderPayments.Calc.Refunds.IntegrationTests.Infrastructure
+{
+    [TestFixture, SetupUkprn]
+    public class GivenAPaymentsHistoryRepository
+    {
+        private HistoricalPaymentsRepository _sut;
+
+        [OneTimeSetUp]
+        public void Setup()
+        {
+            _sut = new HistoricalPaymentsRepository(GlobalTestContext.Instance.TransientConnectionString);
+        }
+
+        [TestFixture, SetupNoPaymentsHistory]
+        public class AndThereAreNoHistoricalPaymentsForProvider : GivenAPaymentsHistoryRepository
+        {
+            [TestFixture]
+            public class WhenCallingGetAllForProvider : AndThereAreNoHistoricalPaymentsForProvider
+            {
+                [Test]
+                public void ThenItReturnsAnEmptyList()
+                {
+                    var result = _sut.GetAllForProvider(RefundsTestContext.Ukprn);
+                    result.Should().BeEmpty();
+                }
+            }
+        }
+
+        [TestFixture, SetupPaymentsHistory]
+        public class AndThereAreSomeHistoricalPaymentsForProvider : GivenAPaymentsHistoryRepository
+        {
+            [TestFixture]
+            public class WhenCallingGetAllForProvider : AndThereAreSomeHistoricalPaymentsForProvider
+            {
+                private List<HistoricalPaymentEntity> _actualHistoricalPayments;
+                private List<HistoricalPaymentEntity> _expectedHistoricalPayments;
+
+                [SetUp]
+                public new void Setup()
+                {
+                    _actualHistoricalPayments = _sut.GetAllForProvider(RefundsTestContext.Ukprn);
```

Comment:
I know both Setup  methods will run, but are we certain the base setup method will always run first?

Reply:
This does need some work to avoid using [SetUp], but not worth holding up this task to sort it.